core/arch/arm/mm/tee_mm.c

```diff
@@ -319,7 +333,17 @@ bool tee_mm_addr_is_within_range(tee_mm_pool_t *pool, paddr_t addr)
 
 bool tee_mm_is_empty(tee_mm_pool_t *pool)
 {
-	return pool == NULL || pool->entry == NULL || pool->entry->next == NULL;
+	bool ret;
+	uint32_t exceptions;
+
+	if (pool == NULL || pool->entry == NULL)
+		return true;
+
+	exceptions = cpu_spin_lock_xsave(&pool->lock);
+	ret = pool->entry == NULL || pool->entry->next == NULL;
```

Comment:
minor: test on `pool->entry == NULL` is useless (but does not arm). `pool->entry` is always allocated as long as the pool is not destroyed.